File: pegasusX/apps/warehouse-app-ios/WarehouseApp/Utilities/WarehousePortalFeature.swift
import Foundation

/// Portal-only surfaces — native apps hand off to warehouse-portal (port 3002).
enum WarehousePortalFeature: String, CaseIterable, Identifiable {
    case register
    case setup
    case profile
    case search

    var id: String { rawValue }

    var title: String {
        switch self {
        case .register: "Register warehouse"
        case .setup: "Warehouse setup"
        case .profile: "Profile"
        case .search: "Global search"
        }
    }

    var subtitle: String {
        switch self {
        case .register: "Create a new warehouse account"
        case .setup: "Location, billing, and configuration"
        case .profile: "Account and warehouse identity"
        case .search: "Jump to any portal page"
        }
    }

    var portalPath: String {
        switch self {
        case .register: "/auth/register"
        case .setup: "/setup/location"
        case .profile: "/profile"
        case .search: "/"
        }
    }

    var systemImage: String {
        switch self {
        case .register: "building.2"
        case .setup: "gearshape.2"
        case .profile: "person.crop.circle"
        case .search: "magnifyingglass"
        }
    }

    var handoffMessage: String {
        switch self {
        case .register:
            "New warehouse registration is completed on the warehouse web portal."
        case .setup:
            "Warehouse setup and onboarding run on the web portal after registration."
        case .profile:
            "Profile and account settings are managed on the warehouse web portal."
        case .search:
            "Global search (⌘K) is available on the warehouse web portal desktop shell."
        }
    }
}

File: pegasusX/apps/warehouse-app-android/app/src/main/java/com/pegasusx/warehouse/ui/portal/WarehousePortalFeature.kt
package com.pegasusx.warehouse.ui.portal

import androidx.compose.ui.res.stringResource

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Search
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Store
import androidx.compose.ui.graphics.vector.ImageVector

/** Portal-only surfaces — native apps hand off to warehouse-portal (port 3002). */
enum class WarehousePortalFeature(
    val routeKey: String,
    val title: String,
    val subtitle: String,
    val portalPath: String,
    val icon: ImageVector,
) {
    REGISTER(
        routeKey = "register",
        title = stringResource(R.string.mobile_warehouse_ui_register_warehouse),
        subtitle = "Create a new warehouse account",
        portalPath = "/auth/register",
        icon = Icons.Default.Store,
    ),
    SETUP(
        routeKey = "setup",
        title = stringResource(R.string.warehouse_portal_setup_setup_wizard_shell_text_warehouse_setup),
        subtitle = "Location, billing, and configuration",
        portalPath = "/setup/location",
        icon = Icons.Default.Settings,
    ),
    PROFILE(
        routeKey = "profile",
        title = stringResource(R.string.portal_nav_profile),
        subtitle = "Account and warehouse identity",
        portalPath = "/profile",
        icon = Icons.Default.Person,
    ),
    SEARCH(
        routeKey = "search",
        title = stringResource(R.string.mobile_warehouse_ui_global_search),
        subtitle = "Jump to any portal page",
        portalPath = "/",
        icon = Icons.Default.Search,
    ),
    ;

    val handoffMessage: String
        get() = when (this) {
            REGISTER -> "New warehouse registration is completed on the warehouse web portal."
            SETUP -> "Warehouse setup and onboarding run on the web portal after registration."
            PROFILE -> "Profile and account settings are managed on the warehouse web portal."
            SEARCH -> "Global search (⌘K) is available on the warehouse web portal desktop shell."
        }

    companion object {
        fun fromRouteKey(key: String): WarehousePortalFeature? =
            entries.firstOrNull { it.routeKey == key }
    }
}
